plugins/modules/notification/mattermost.py

```diff
@@ -144,7 +157,7 @@ def main():
 
     # define payload
     payload = {}
```

Comment:
The code somehow **must** use `https://www.ansible.com/favicon.ico` for the `icon_url` if nothing is specified. Otherwise this is a backwards-compatibility breaking change and cannot be merged.

Reply:
https://docs.mattermost.com/developer/webhooks-incoming.html

> When using the icon_emoji parameter, the user profile image is replaced by the emoji provided. This will also 
> override the icon_url parameter if both are provided.

We could just set the default for icon_url again and if icon_emoji is set, it will override any given URL (also the default value).
I still think that icon_url should not have a default value, because it's an optional parameter after all and if not specified, the icon the webhook creator specified for the webhook will be used and if there is none, the icon of the webhook creator will be used, so leaving this option out will not really break something, it will just make the webhook notifications look differently because the redhat icon is not there anymore.

If you consider this as a breaking change and still want me to default the icon_url to the previous domain, I'm happy to submit another patch doing just that.

Reply:
> it will just make the webhook notifications look differently because the redhat icon is not there anymore.

That is a breaking change: behavior will be different when the default is removed. I agree that having this default value probably wasn't the best idea, but that has been decided in the past and now we have it, and so we have to keep it to avoid breaking the assumptions of existing users of the module.

So the default has to stay, either as part of documentation and set in the code, or as an explicit default value handled by Ansible. Keeping it should be fine by what you wrote above though:

> https://docs.mattermost.com/developer/webhooks-incoming.html
> 
> > When using the icon_emoji parameter, the user profile image is replaced by the emoji provided. This will also
> > override the icon_url parameter if both are provided.

Reply:
I've updated the documentation a bit and have also fixed other parts of it - let me know what you think. I'm not happy with the two examples for icon_url and icon_emoji, I'd like to combine that into one, but am not sure about the syntax to use for that so that it's clear to the user.